base/linalg/transpose.jl

```diff
@@ -0,0 +1,334 @@
+immutable Transpose{T,S<:AbstractMatrix} <: AbstractMatrix{T}
+    data::S
+    conjugated::Bool
+end
```

Comment:
@stevengj , thanks for educating me. I knew my background was probably to specific.

Anyway, I am fine with any approach and especially like the idea of @timholy and @mbauman. I just don't understand why BLAS did implement transpose and ctranspose but then not plain conjugated to make the picture complete.